url82787aa7a4_vrthra_v_language_tgz-pJ8-v_VJ8/src/v/Token.java

```diff
@@ -0,0 +1,81 @@
+package v;
+
+@SuppressWarnings ({"unchecked"})
+public abstract class Token {
+    public abstract String value();
+    public abstract Type type();
+
+    public boolean bvalue() {
+        try {
+            Term<Boolean> v = (Term<Boolean>)this;
+            return v.val;
+        } catch (Exception e) {
+            throw new VException("err:type:bool",this,"Invalid type(need bool)");
+        }
+    }
+    public char cvalue() {
+        try {
+            Term<Character> v = (Term<Character>)this;
+            return v.val;
+        } catch (Exception e) {
+            throw new VException("err:type:bool",this,"Invalid type(need bool)");
+        }
+    }
+    public int ivalue() {
+        try {
+            Term<Integer> v = (Term<Integer>)this;
+            return v.val;
+        } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)